src/classes/PS_CommitmentRequest_TEST.cls

```diff
@@ -33,7 +33,7 @@
 * @group Elevate Payment Services
 * @description Tests verifying Elevate payment services commitment request
 */
-@IsTest(IsParallel=true)
```

Comment:
The `PS_CommitmentRequest_TEST` was recently created but it was re-run in most builds. This is not expected since the test class is relatively small (three tests) and it is using `testSetup()` method to create required data for all tests. So, this was a lucky break to realize that not necessarily we need to change RD2 test classes to avoid DMLs, but also chance parallel testing attribute to `false`. After changing the parallel attribute, none of the changed classes is re-run, therefore, the total build time is decreased.

So based on one [build](https://sfdo-metaci.herokuapp.com/builds/342561/flows) in another PR, the re-run was applied to "Retrying 263 failed methods from 19 test classes" and took about 21 min.
Another random chosen [build](https://sfdo-metaci.herokuapp.com/builds/338372/flows) had "Retrying 342 failed methods from 19 test classes" and took 59 min.
Another PR random [build](https://sfdo-metaci.herokuapp.com/builds/342549/flows) had "Retrying 305 failed methods from 19 test classes" and took 30 min.

A [build](https://sfdo-metaci.herokuapp.com/builds/342016/flows) in *this PR* has "Retrying 116 failed methods from 17 test classes" and it took 40 min.
Another [build](https://sfdo-metaci.herokuapp.com/builds/342461/flows) had "Retrying 79 failed methods from 13 test classes which took 6min.

So even though we cannot influence how long other tests re-runs will take, we did decrease number of re-run tests and increased the probability of builds running faster. Based on builds in this PR, time to run tests (excluding installed package deployments and post deploy cleanup) takes from 1h 10min to 1h 20min.

CC @force2b , @andrewyu-salesforce , @npsp-reedestockton , @ptewson-sfdo

Reply:
Great catch @RadSFDeveloper